### ISSUE TITLE
Assertion failed with Gerrit-1.2.3-146 / IntelliJ IDEA 2020.3 IC-203.5981.155

### ISSUE THREAD

**Issue by bojanantonovic** [uwolfer/gerrit-intellij-plugin#399]
Since some days IntelliJ always reports with a red question mark:
```
java.lang.Throwable: Assertion failed
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:195)
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:204)
	at org.jetbrains.ide.BuiltInServerManagerImpl.waitForStart(BuiltInServerManagerImpl.kt:109)
	at org.jetbrains.git4idea.ssh.GitXmlRpcHandlerService.getXmlRcpPort(GitXmlRpcHandlerService.java:71)
	at git4idea.commands.GitHandlerAuthenticationManager.prepareHttpAuth(GitHandlerAuthenticationManager.java:93)
	at git4idea.commands.GitHandlerAuthenticationManager.lambda$prepare$0(GitHandlerAuthenticationManager.java:60)
	at git4idea.GitUtil.tryRunOrClose(GitUtil.java:1041)
	at git4idea.commands.GitHandlerAuthenticationManager.prepare(GitHandlerAuthenticationManager.java:59)
	at git4idea.commands.GitImplBase.run(GitImplBase.java:171)
	at git4idea.commands.GitImplBase.runCommandWithoutCollectingOutput(GitImplBase.java:105)
	at git4idea.history.GitDetailsCollector.readRecordsFromHandler(GitDetailsCollector.kt:95)
	at git4idea.history.GitDetailsCollector.readFullDetailsFromHandler(GitDetailsCollector.kt:79)
	at git4idea.history.GitDetailsCollector.readFullDetails(GitDetailsCollector.kt:32)
	at git4idea.history.GitLogUtil.readFullDetails(GitLogUtil.java:197)
	at git4idea.history.GitHistoryUtils.history(GitHistoryUtils.java:145)
	at com.urswolfer.intellij.plugin.gerrit.ui.RepositoryChangesBrowserProvider$GerritRepositoryChangesBrowser.getCommit(RepositoryChangesBrowserProvider.java:225)
	at com.urswolfer.intellij.plugin.gerrit.ui.RepositoryChangesBrowserProvider$GerritRepositoryChangesBrowser.access$1000(RepositoryChangesBrowserProvider.java:110)
	at com.urswolfer.intellij.plugin.gerrit.ui.RepositoryChangesBrowserProvider$GerritRepositoryChangesBrowser$4.call(RepositoryChangesBrowserProvider.java:191)
	at com.urswolfer.intellij.plugin.gerrit.ui.RepositoryChangesBrowserProvider$GerritRepositoryChangesBrowser$4.call(RepositoryChangesBrowserProvider.java:184)
	at com.urswolfer.intellij.plugin.gerrit.git.RevisionFetcher$FetchCallback.call(RevisionFetcher.java:114)
	at com.urswolfer.intellij.plugin.gerrit.git.RevisionFetcher$FetchCallback.call(RevisionFetcher.java:99)
	at com.urswolfer.intellij.plugin.gerrit.git.GerritGitUtil$1.onSuccess(GerritGitUtil.java:153)
	at com.intellij.openapi.progress.impl.CoreProgressManager.finishTask(CoreProgressManager.java:578)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.finishTask(ProgressManagerImpl.java:154)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsync$6(CoreProgressManager.java:500)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:216)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:24)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:199)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:324)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:85)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:134)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:976)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:501)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by uwolfer** [comment]
Thanks for creating an issue for that. Probably 2 days ago, this exception started to appear for a lot of users. Looks related to latest IntelliJ update:

IntelliJ version: IntelliJ IDEA 2020.3 IC-203.5981.155
Java: JetBrains s.r.o. 11.0.9

So far, I've no idea what causes this issue, but at least from the trace, it looks like somethings goes wrong in an IntelliJ component.

Does this exception cause any noticeable issues other than the red question mark?

**Comment by bojanantonovic** [comment]
I haven't realized anything disturbing. I can still work with the Geriet plugin as usual.

**Comment by leonid-s-usov** [comment]
It's amazing that the IDE has thrown this message to me. Not sure if it's gerrit specific or (most probably) IDE driven, but still impressive.
Anyhow, great plugin
![image](https://user-images.githubusercontent.com/3144935/101096717-a43b1980-35c8-11eb-8a23-9e338e86fd61.png)

**Comment by uwolfer** [comment]
> It's amazing that the IDE has thrown this message to me. Not sure if it's gerrit specific or (most probably) IDE driven, but still impressive.

@leonid-s-usov: This dialog is specific for this plugin when you report an exception for IDE. I've enabled it yesterday after I've received that many exceptions with all the same error; I've added such a functionality to the plugin a long time ago ([code](https://github.com/uwolfer/gerrit-intellij-plugin/blob/intellij14/src/main/java/com/urswolfer/intellij/plugin/gerrit/errorreport/PluginErrorReportSubmitter.java#L91-L94)).

**Comment by lejsekt** [comment]
It happens in Rider 2020.3.2 as well.

**Comment by StefanPenndorf** [comment]
Looks like a duplicate of https://github.com/uwolfer/gerrit-intellij-plugin/issues/376 which reads "`GitHistoryUtils.history` is called from the EDT, leading to an assertion error". Stacktraces look almost identical (different line numbers in some classes but same methods called). Interestingly #376 has been reported in February 2020 already.

I just have to join the others: Great plugin, thanks for the effort @uwolfer . Unfortunately, the error message is a bit annoying, even if I haven't noticed any problems with the Gerrit plugin myself. What information do you need for a further error analysis? How can I help?

**Comment by uwolfer** [comment]
I've just released a new version (v1.2.4) which should fix these issues, you should find the update in IntelliJ plugin manager. Thanks for your patience.
